spec/lib/preservation_catalog/s3_spec.rb

```diff
@@ -21,26 +21,81 @@
   end
 
   describe 'config' do
-    let(:config) { described_class.client.config }
-    let(:trio) do
-      {
-        'AWS_SECRET_ACCESS_KEY' => 'secret',
-        'AWS_ACCESS_KEY_ID' => 'some_key',
-        'AWS_REGION' => 'us-east-1'
-      }
-    end
+    context 'with access key and region env vars' do
+      let(:config) { described_class.client.config }
+      let(:trio) do
+        {
+          'AWS_SECRET_ACCESS_KEY' => 'secret',
+          'AWS_ACCESS_KEY_ID' => 'some_key',
+          'AWS_REGION' => 'us-east-1'
+        }
+      end
 
-    around do |example|
-      old_vals = trio.keys.zip(ENV.values_at(*trio.keys)).to_h
-      trio.each { |k, v| ENV[k] = v }
-      example.run
-      old_vals.each { |k, v| ENV[k] = v }
+      around do |example|
+        old_vals = trio.keys.zip(ENV.values_at(*trio.keys)).to_h
+        trio.each { |k, v| ENV[k] = v }
+        example.run
+        old_vals.each { |k, v| ENV[k] = v }
+      end
+      it 'pulls from ENV vars' do
+        expect(config.region).to eq 'us-east-1'
+        expect(config.credentials).to be_an(Aws::Credentials)
+        expect(config.credentials).to be_set
+        expect(config.credentials.access_key_id).to eq 'some_key'
+      end
     end
-    it 'pulls from ENV vars' do
-      expect(config.region).to eq 'us-east-1'
-      expect(config.credentials).to be_an(Aws::Credentials)
-      expect(config.credentials).to be_set
-      expect(config.credentials.access_key_id).to eq 'some_key'
+
+    context 'pointing the client to shared credentials' do
+      let(:config) { described_class.client.config }
+      let(:shared_credentials) do
+        Aws::SharedCredentials.new(path: Rails.root.join('spec', 'fixtures', 'aws_credentials'))
+      end
+
+      after do
+        Aws.config = {}
+      end
+
+      context 'profile us_west_2' do
+        let(:uno) do
+          {
+            'AWS_PROFILE' => 'us_west_2'
+          }
+        end
+
+        around do |example|
+          old_vals = uno.keys.zip(ENV.values_at(*uno.keys)).to_h
+          uno.each { |k, v| ENV[k] = v }
+          example.run
+          old_vals.each { |k, v| ENV[k] = v }
+        end
+
+        it 'pulls the one profile from a config file' do
+          Aws.config.update(region: 'us-west-2', credentials: shared_credentials)
+          expect(config.region).to eq 'us-west-2'
+          expect(config.credentials.credentials.access_key_id).to eq 'foo'
+          expect(config.credentials.credentials.secret_access_key).to eq 'bar'
+        end
+      end
+      context 'profile us_east_1' do
+        let(:uno) do
+          {
+            'AWS_PROFILE' => 'us_east_1'
+          }
```

Comment:
this is a place where you could do

```ruby
let(:uno) { 'AWS_PROFILE' => 'us_east_1' }
```
I believe

Reply:
rspec doesn't like that, but it will accept

```
let(:uno) { { 'AWS_PROFILE' => 'us_west_2' } }
```

I'm happy to change it to a one-liner that way, or leave it as is...

Reply:
meh.

Reply:
I prefer concision, but it isn't critical here.  `let(:uno) { Hash['AWS_PROFILE' => 'us_east_1'] }` works too, if you don't like the double curlies.  (Double curlies are fine by me.)  I might be tempted to rename all these (including the old `trio` as `envs`) since we treat them the same way.